Adding PCB Floorplan and layout from the PSLab Mini Board

Tejasgarg opened the pull request:
- fixes: #16 

The board is set to be 85mm X 56mm with a corner radius of 3mm, to be the same as the Raspberry Pi.

The Mounting Holes are set to have a 2.75mm inner diameter to also match the Raspberry Pi and have their centres positioned the same as the Raspberry Pi mounting holes.

The Header Pins are placed 3.5mm away from the edge, also matching the Raspberry Pi.

Feducials are kept at 2mm away from the edge, same as the current PSLab board.

Now, for the ports, I have ensured that they are compatible with all the series of Raspberry Pi encasings (Raspberry Pi 3, 4 and 5)
This can be visualised using the following Drawing - 

<img width="1048" height="612" alt="Screenshot 2025-08-08 041756" src="https://github.com/user-attachments/assets/1d423db2-c8cc-4173-83db-d9a210ae1fa9" />

I have arranged all the ports such that the new board will be able to fit into any of the enclosures from the Raspberry Pi series of boards.

For the Battery Connector Port
The side USB connector is consistent in all the Raspberry PI, so I used that as the exit point for the connector.

The following diagram visualises it - 
<img width="893" height="718" alt="Screenshot 2025-08-08 223723" src="https://github.com/user-attachments/assets/933124fb-7757-4d06-b700-ab563795bfda" />

Since it was not possible for the battery connector to have a universal placement where it was compliant with all the models of the Raspberry Pi Series enclosures, I have placed it 3.5mm from the edge, so that the connector, along with the input port, can completely fit inside the enclosure, and the wire can exit from the port cutout.

Also placed other components like the main processor, ESP circuitry and battery management chip.

I will continue working on placing the rest of the components and their accompanying resistors and capacitors, along with trying to optimise the placement of the currently placed components.

CloudyPadmal commented:
I see that the layout is missing some of the 3D parts for models such as the BNC. Can you please find 3D parts and update?

Tejasgarg commented:
Sure!
I will add the 3d model for the parts that are missing it.

Tejasgarg commented:
Hey @CloudyPadmal!
Updated the PR and now every component on the board has a 3d model.
Also changed the layout with pin assignment and placement changes for better and cleaner routing.
Kindly review whenever you get time :)

CloudyPadmal reviewed (changes requested):
1. Remove `FID2` next to chip antenna and move it to top right of the board.
2. Hide silk screen markings of `FID?` and `MH?`
3. Silk screen text are overlapping with courtyard. Fix those.
4. There are some weird traces next to U3 region (136,93) to (138,96)
5. Silk screen text size is too thick. Please refer to DRC errors. (I will update the github actions to reflect DRC and ERC errors)
6. SPI Flash is too far from the ESP IC.
7. There is too much clutter from oscilloscope probe input to MCU pins.
8. ESP protection diodes (U4) is too far from the USB C connector.
9. Use 0402 resistors for LED resistors.
10. Placement of `D7-D10` LEDs block clean routing for USB data lines.
11. Test points for USB lines have large silk screen text.
12. Crystals need to be very close to the pins. 

Overall, the traces need to be short and direct as much as possible without routing all over the board.

Tejasgarg commented:
> 1. Remove `FID2` next to chip antenna and move it to top right of the board.
> 2. Hide silk screen markings of `FID?` and `MH?`
> 3. Silk screen text are overlapping with courtyard. Fix those.
> 4. There are some weird traces next to U3 region (136,93) to (138,96)
> 5. Silk screen text size is too thick. Please refer to DRC errors. (I will update the github actions to reflect DRC and ERC errors)
> 6. SPI Flash is too far from the ESP IC.
> 7. There is too much clutter from oscilloscope probe input to MCU pins.
> 8. ESP protection diodes (U4) is too far from the USB C connector.
> 9. Use 0402 resistors for LED resistors.
> 10. Placement of `D7-D10` LEDs block clean routing for USB data lines.
> 11. Test points for USB lines have large silk screen text.
> 12. Crystals need to be very close to the pins.
> 
> Overall, the traces need to be short and direct as much as possible without routing all over the board.

Hey @CloudyPadmal!
I have tried to address all the changes requested 
Though I wanted to clarify something about point 4, the traces are NetTies without which the ERC gives an issue "Pins of type Power output and Power output are connected". Is there any another, better way to go about solving this error?

Also getting another error in DRC regarding clearance violation, where a NPTH is placed at less than 0.2mm, in the case of holes for USB-C. 
I tried checking design constraints from popular production plants like JLCPCB, as mentioned on their website, track to NPTH minimum distance at 0.2mm
![Screenshot 2025-08-15 at 1 23 02 AM](https://github.com/user-attachments/assets/e754d1cd-990c-428e-ae50-06fb32097f87)

But most of the popular USB-C jacks have similar holes for mounting onto the PCB, so what could be done about this?
Should I go ahead and reduce the constraint, or is there another workaround for this that I am missing?

CloudyPadmal reviewed (changes requested):
I don't see any fix for Oscilloscope traces being routed all over the board. 

In top right, the oscilloscope input needs to travel across voltage doubler/inverter circuits. These will ruin the OSC signals. Have a look at PSLab V6 layout where the OpAmps condition those OSC inputs and change the placement of components accordingly.

ESP antenna trace is also very far from the antenna input pin.

Check the whole board to improve the placement of critical traces.

CloudyPadmal reviewed (approved):
Now it looks good to go ahead with!
